lib/mix/lib/mix/tasks/app.tree.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Mix.Tasks.App.Tree do
+  use Mix.Task
+
+  @shortdoc "Analyses the current Mix project and prints dependency tree on terminal."
+
+  @moduledoc """
+  Analyses the current Mix project and prints dependency tree on terminal.
+
+  ## Command line options
+
+    * `--exclude` - exclude dependencies, which you do not want to see in a tree
+
+  ## Example of use
+
+      mix deps.tree --exclude logger,elixir
```

Comment:
I think we should implement `--exclude` option without `String.split/2` but by using `:keep`, that's what we do for `mix test` `--exclude` and `--include` options.